src/uniswapMappings/uniswapPools.ts

```diff
@@ -0,0 +1,143 @@
+
+  // update ETH price now that prices could have changed
+  let bundle = Bundle.load('1')
+  bundle.ethPriceUSD = getEthPriceInUSD()
+  bundle.save()
+
+  // update token prices
+  token0.derivedETH = findEthPerToken(token0 as Token)
+  token1.derivedETH = findEthPerToken(token1 as Token)
+  token0.derivedUSD = token0.derivedETH.times(bundle.ethPriceUSD)
+  token1.derivedUSD = token1.derivedETH.times(bundle.ethPriceUSD)
+  token0.save()
+  token1.save()
+}
+
+export function handleMint(event: MintEvent): void {
+  let poolAddress = event.address.toHexString()
+  let pool = UniswapPool.load(poolAddress)
+
+  let token0 = Token.load(pool.token0)
+  let token1 = Token.load(pool.token1)
+  let amount0 = convertTokenToDecimal(event.params.amount0, BigInt.fromI32(token0.decimals))
+  let amount1 = convertTokenToDecimal(event.params.amount1, BigInt.fromI32(token1.decimals))
+
+  // Pools liquidity tracks the currently active liquidity given pools current tick.
+  // We only want to update it on mint if the new position includes the current tick.
+  if (
+    pool.tick !== null &&
+    BigInt.fromI32(event.params.tickLower).le(pool.tick as BigInt) &&
+    BigInt.fromI32(event.params.tickUpper).gt(pool.tick as BigInt)
+  ) {
+    pool.liquidity = pool.liquidity.plus(event.params.amount)
+  }
+
+  pool.totalValueLockedToken0 = pool.totalValueLockedToken0.plus(amount0)
+  pool.totalValueLockedToken1 = pool.totalValueLockedToken1.plus(amount1)
+
+  // TODO: Update Tick's volume, fees, and liquidity provider count. Computing these on the tick
+  // level requires reimplementing some of the swapping code from v3-core.
+
+  token0.save()
+  token1.save()
+  pool.save()
+
+}
+
+export function handleBurn(event: BurnEvent): void {
+  let poolAddress = event.address.toHexString()
+  let pool = UniswapPool.load(poolAddress)
+
+  let token0 = Token.load(pool.token0)
+  let token1 = Token.load(pool.token1)
+  let amount0 = convertTokenToDecimal(event.params.amount0, BigInt.fromI32(token0.decimals))
+  let amount1 = convertTokenToDecimal(event.params.amount1, BigInt.fromI32(token1.decimals))
+
+  // Pools liquidity tracks the currently active liquidity given pools current tick.
+  // We only want to update it on burn if the position being burnt includes the current tick.
+  if (
+    pool.tick !== null &&
+    BigInt.fromI32(event.params.tickLower).le(pool.tick as BigInt) &&
+    BigInt.fromI32(event.params.tickUpper).gt(pool.tick as BigInt)
+  ) {
+    pool.liquidity = pool.liquidity.minus(event.params.amount)
+  }
+
+  pool.totalValueLockedToken0 = pool.totalValueLockedToken0.minus(amount0)
+  pool.totalValueLockedToken1 = pool.totalValueLockedToken1.minus(amount1)
+
+  token0.save()
+  token1.save()
+  pool.save()
+}
+
+export function handleSwap(event: SwapEvent): void {
+  let bundle = Bundle.load('1')
+  let pool = UniswapPool.load(event.address.toHexString())
+
+  // hot fix for bad pricing
+  if (pool.id == '0x9663f2ca0454accad3e094448ea6f77443880454') {
```

Comment:
magic number: can you create a good name for this and create a const. Also comment is not clear to me

Reply:
Makes sense. It seems to be a hotfix on UniV3 subgraph.